build-tools/packages/build-cli/src/library/commands/generateSourceEntrypoints.ts

```diff
@@ -0,0 +1,336 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import fs from "node:fs/promises";
+
+import type { PackageJson } from "@fluidframework/build-tools";
+import { Flags } from "@oclif/core";
+import type { ExportSpecifierStructure, Node } from "ts-morph";
+import { ModuleKind, Project, ScriptKind } from "ts-morph";
+
+import type { CommandLogger } from "../../logging.js";
+import { BaseCommand } from "./base.js";
+
+import { ApiLevel } from "../apiLevel.js";
+import { ApiTag } from "../apiTag.js";
+import type { ExportData } from "../packageExports.js";
+import { queryDefaultResolutionPathsFromPackageExports } from "../packageExports.js";
+import { getApiExports, getPackageDocumentationText } from "../typescriptApi.js";
+
+import type { TsConfigJson } from "type-fest";
+
+const optionDefaults = {
+	mainEntrypoint: "./src/index.ts",
+	outFileAlpha: ApiLevel.alpha,
+	outFileBeta: ApiLevel.beta,
+	outFileLegacy: ApiLevel.legacy,
+	outFilePublic: ApiLevel.public,
+	outFileSuffix: ".ts",
+	srcDir: "src/",
+} as const;
+
+/**
+ * Generates type declarations files for Fluid Framework APIs to support API levels (/alpha, /beta. etc.).
+ */
+export class GenerateSourceEntrypointsCommand extends BaseCommand<
+	typeof GenerateSourceEntrypointsCommand
+> {
+	static readonly description =
+		`Generates type declaration entrypoints for Fluid Framework API levels (/alpha, /beta. etc.) as found in package.json "exports"`;
+
+	static readonly flags = {
+		mainEntrypoint: Flags.file({
+			description: "Main entrypoint file containing all untrimmed exports.",
+			default: optionDefaults.mainEntrypoint,
+			exists: true,
+		}),
+		outFileAlpha: Flags.string({
+			description: "Base file name for alpha entrypoint declaration files.",
+			default: optionDefaults.outFileAlpha,
+		}),
+		outFileBeta: Flags.string({
+			description: "Base file name for beta entrypoint declaration files.",
+			default: optionDefaults.outFileBeta,
+		}),
+		outFileLegacy: Flags.string({
+			description: "Base file name for legacy entrypoint declaration files.",
+			default: optionDefaults.outFileLegacy,
+		}),
+		outFilePublic: Flags.string({
+			description: "Base file name for public entrypoint declaration files.",
+			default: optionDefaults.outFilePublic,
+		}),
+		outFileSuffix: Flags.string({
+			description:
+				"File name suffix including extension for emitting entrypoint declaration files.",
+			default: optionDefaults.outFileSuffix,
+		}),
```

Comment:
We can skip all of these options. We don't even use them with the other command. If we need options we can complicate things later.